llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -668,40 +669,17 @@ static string_vector saveResultSymbolsLists(string_vector &ResSymbolsLists,
     }                                                                          \
   }
 
-// Helper function for creating Inliner pass.
-// The approach is taken from opt tool.
-static Pass *createFunctionInliningPassHelper() {
-  if (OptLevelO0)
-    return createFunctionInliningPass(0, 0, false);
-
-  if (OptLevelO1)
-    return createFunctionInliningPass(1, 0, false);
-
-  if (OptLevelO2)
-    return createFunctionInliningPass(2, 0, false);
-
-  if (OptLevelOs)
-    return createFunctionInliningPass(2, 1, false);
-
-  if (OptLevelOz)
-    return createFunctionInliningPass(2, 2, false);
-
-  if (OptLevelO3)
-    return createFunctionInliningPass(3, 0, false);
-
-  return createFunctionInliningPass();
-}
-
 // When ESIMD code was separated from the regular SYCL code,
 // we can safely process ESIMD part.
 // TODO: support options like -debug-pass, -print-[before|after], and others
 static void LowerEsimdConstructs(Module &M) {
   legacy::PassManager MPM;
   MPM.add(createSYCLLowerESIMDPass());
   if (!OptLevelO0) {
-    // Inlining and SROA passes are required to make
-    // ESIMD/accessor_gather_scatter.cpp test work.
-    MPM.add(createFunctionInliningPassHelper());
+    // LowerESIMD pass sets 'alwaysinline' attribute to all the functions
+    // in the call graph for every ESIMD kernel, which is later consumed
+    // by AlwaysInliner pass.
+    MPM.add(createAlwaysInlinerLegacyPass());
```

Comment:
@robertcox-github, is it a fair assumption, that there is no point in scheduling a regular inliner pass if we set `alwaysinline` attribute for every potentially inlinable function and process it with AlwaysInliner pass? I.e. there will be no inlining opportunities for a regular inliner pass.